README.md

```diff
@@ -26,79 +26,173 @@ pip uninstall securedrop-sdk
+
+### Generating cassettes for API calls over HTTP
+
+1. Start the server in a docker container by running:
+
+    ```bash
+    NUM_SOURCES=5 make dev
+    ```
+
+2. [Skip if adding a new test] Delete the cassettes you wish to regenerate or just delete all yaml files by running:
+
+    ```bash
+    rm data/*.yml
+    ```
+
+3. Generate new cassettes that make API calls over HTTP by running:
+
+    ```bash
+    make test TESTS=tests/test_api.py
+    ```
+Note: Some tests alter source and conversation data on the server so you may need to restart the server in between test runs.
+
+### Generating cassettes for API calls over qrexec
+
+In order to generate cassettes for tests that make API calls over qrexec, you'll need to run the server and proxy on a separate VM. If this is the first time you are generating cassettes, then you'll need to first follow the steps outlined in the [Setup for qrexec communication](#set-up-for-qrexec-communication) section.
+
+Once your VMs are set up, follow these steps:
+
+1. Start the server in a docker container on a separate VM (called `dev-server` if you followed the instructions from [Setup for qrexec communication](#set-up-for-qrexec-communication)) by running:
+
+    ```bash
+    NUM_SOURCES=5 make dev
+    ```
+
+2. [Skip if adding a new test] Delete the cassettes you wish to regenerate or just delete all json files by running:
+
+    ```bash
+    rm data/*.json
+    ```
+
+3. Comment out the `@dastollervey_datasaver` decorator above the test you want to generate a new cassette for or just generate all new cassettes by commenting out the decorator above all methods in the `test_apiproxy.py::TestAPIProxy` class.
+
+4. Make qrexec calls to the server and collect real response data:
+
+    ```bash
+    make test TESTS=tests/test_apiproxy.py
+    ```
+
+5. Uncomment the `@dastollervey_datasaver` decorator wherever you commented it out.
+6. Record new cassettes from the response data collected in step 4:
+
+    ```bash
+    make test TESTS=tests/test_apiproxy.py
+    ```
+**Note:** If you get a 403 error it's becuase the test is trying to reuse an old TOTP code, so wait for 60 seconds and try again. Some tests alter source and conversation data on the server so you should restart the server in between test runs.
+
+**Note:** Remember that file download checks don't read actual file path in the `APIProxy` tests as it requires QubesOS setup. You can manually uncomment those lines to execute them on QubesOS setup.
+
+#### Setting up qrexec communication for SDK test VMs
+
+If this is the first time you are generating new cassettes that make API calls over qrexec, then you'll need to set up a new VM for running the server and proxy following these steps:
+
+1. Create a new template cloned from the **debian-10** template called **dev-server-template**.
+2. Install the lastest proxy package:
+
+    ```bash
+    wget https://apt.freedom.press/pool/main/s/securedrop-proxy/<latest-package>.deb
```

Comment:
`wget` will fail in the template vm, correct?

Reply:
i updated this so that we don't recommend maintaining your own template (even though this is what i do). so now we say to run wget to get the latest proxy package in the appvm directly.